app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/HotspotService.java

```diff
@@ -115,19 +123,62 @@
     return serviceBinder;
   }
 
+  private void setUpHotspotStateReceiver() {
+    hotspotStateReceiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if ("android.net.wifi.WIFI_AP_STATE_CHANGED".equals(action)) {
```

Comment:
use the constant `WifiManager.WIFI_AP_STATE_CHANGED_ACTION` 
Also turn this broadcastReceiver into its own class and communicate back to the service with callbacks

Reply:
Fixed.